src/test/scala/uk/gov/nationalarchives/api/update/ResultCollectorTest.scala

```diff
@@ -1,64 +1,103 @@
+    val sqsUtilsMock = mock[SQSUtils]
+    val circeError = new Exception("decoderError")
+    val responseProcessor = ResultCollector(config, sqsUtilsMock)
+    val updateException = FailedApiUpdateException("receiptHandle", circeError)
+
+    val decoderError: Either[FailedApiUpdateException, Future[String]] = Left(updateException)
     val caught = responseProcessor.collect(List(decoderError)).failed.futureValue
+    verify(sqsUtilsMock, times(1)).makeMessageVisible(queueUrl, "receiptHandle")
     caught.getMessage should include("decoderError")
   }
 
+  "The collect method" should "group errors from a mixture of results and reset messages visibility to allow retry" in {
+    val sqsUtilsMock = mock[SQSUtils]
 
-
-  "The collect method" should "group errors from a mixture of results" in {
-    val circeError1 = mock[circe.Error]
-    when(circeError1.getMessage).thenReturn("first decoder error")
-    val circeError2 = mock[circe.Error]
-    when(circeError2.getMessage).thenReturn("second decoder error")
+    val circeError1 = new Exception("first decoder error")
+    val circeError2 = new Exception("second decoder error")
     val apiError1 = new RuntimeException("first API error")
     val apiError2 = new HttpException(Response(Left("second API error"), StatusCode.InternalServerError))
+
+    val updateExCirceError1 = FailedApiUpdateException("receiptHandle1", circeError1)
+    val updateExCirceError2 = FailedApiUpdateException("receiptHandle2", circeError2)
+    val updateExApiError1 = FailedApiUpdateException("receiptHandle3", apiError1)
+    val updateExApiError2 = FailedApiUpdateException("receiptHandle4", apiError2)
+
     val errors = List(
-      Left(circeError1),
-      Right(Future.failed(apiError1)),
+      Left(updateExCirceError1),
+      Right(Future.failed(updateExApiError1)),
       Right(Future.successful("Successful result")),
-      Left(circeError2),
-      Right(Future.failed(apiError2))
+      Left(updateExCirceError2),
+      Right(Future.failed(updateExApiError2))
     )
 
-    val responseProcessor = ResultCollector()
+    val responseProcessor = ResultCollector(config, sqsUtilsMock)
     val caught = responseProcessor.collect(errors).failed.futureValue
 
+    verify(sqsUtilsMock, times(1)).makeMessageVisible(queueUrl, "receiptHandle1")
+    verify(sqsUtilsMock, times(1)).makeMessageVisible(queueUrl, "receiptHandle2")
+    verify(sqsUtilsMock, times(1)).makeMessageVisible(queueUrl, "receiptHandle3")
+    verify(sqsUtilsMock, times(1)).makeMessageVisible(queueUrl, "receiptHandle4")
+
     caught.getMessage should include("first decoder error")
     caught.getMessage should include("second decoder error")
     caught.getMessage should include("first API error")
     caught.getMessage should include("second API error")
   }
 
+  "The collect method" should "not reset message visibility for non failed api update exception" in {
```

Comment:
Do you need this test. I'm not sure if you ever could get a non `FailedApiUpdateException` because you're calling recover and [wrapping any API update failures here](https://github.com/nationalarchives/tdr-api-update/pull/21/files#diff-0f9551c5dc77e5c605a195fe9aebcf366dd7ac6ff0f7e6c2b261350a074ffb57R62)

I'm not 100% sure but if there's no way of getting an unwrapped exception in here then you can get rid of this test.

Reply:
Fair point.